### PR TITLE
Only load live ClientAccounts when creating/editing an auction.

### LINKED ISSUE #1246: Only show live Tock projects in "Billable to" dropdown
So that I don't accidentally assign an expired project to a new auction, I should only see Tock projects that are "live" (that are on or after the project start date, and on or before the project end date) when I am creating an auction.

- [ ] **Scenario: Admin can only assign live Tock projects to new auction**
- Given I am an authenticated admin
- And I'm creating an auction (https://micropurchase.18f.gov/admin/auctions/new)
- When I click the dropdown for "Billable to"
- Then I should only see Tock projects that are "live" (see above)

we should make sure that only active tock projects show up on dropdown, but we should not delete old tock projects since they might be associated with existing auctions (closed) and should still be displayed

### PR DESCRIPTION
**Pull request by adelevie**
closes #1246

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -73,6 +72,7 @@
     t.string   "name",       null: false
     t.boolean  "billable",   null: false
     t.integer  "tock_id",    null: false
+    t.boolean  "live"
```
Comment (jessieay) [inline db/schema.rb]: default for this?

**Comment by jessieay** [comment]
I feel like there should be some updates to the TockImporter that updates ClientAccounts' `live` status? Might be easier to just save `start_date` and `end_date` from Tock API instead and only show items that are "live" based on those dates rather than using a boolean

**Comment by adelevie** [comment]
why is it better for micro-purchase to calculate "live"-ness if Tock already is doing that? Or could we have a scenario where live == false?

File: db/schema.rb (hunk 1 above)
Comment (adelevie) [inline db/schema.rb]: I guess we could make it false? But we're getting this value from Tock, so I am unsure.

Comment (jessieay) [inline db/schema.rb]: Oh I didn't realize that. Where in this PR are we getting that from Tock?

File: app/services/tock_importer.rb [hunk 2]
```diff
@@ -6,6 +6,7 @@ def perform
       tock_project = ClientAccount.find_or_initialize_by(tock_id: project["id"])
       tock_project.name = project["name"]
       tock_project.billable = project["billable"]
+      tock_project.live = project["live"]
```
Comment (jessieay) [inline app/services/tock_importer.rb]: oh here it is! not sure how I missed that before

File: db/schema.rb (hunk 1 above)
Comment (jessieay) [inline db/schema.rb]: yeah let's default to `false` and then let tock tell us when things are live

File: app/services/tock_importer.rb (hunk 2 above)
Comment (jessieay) [inline app/services/tock_importer.rb]: looks like tock calls this `"active"` -- should we do same?

Comment (jessieay) [inline app/services/tock_importer.rb]: spec for this importer update?

**Comment by jessieay** [comment]
This is looking super close, @adelevie !

File: app/services/tock_importer.rb (hunk 2 above)
Comment (adelevie) [inline app/services/tock_importer.rb]: more than this? https://github.com/18F/micropurchase/blob/cea84c1dad7f1c11006005bae860f22f7c740065/spec/services/tock_importer_spec.rb

File: spec/support/fixtures/tock_projects.json [hunk 3]
```diff
@@ -1 +1,37 @@
...
+    },
+    {
+      "id": 96,
+      "client": "General Services Administration - 18F (Non-Billable)",
+      "name": "18F Guides",
+      "description": " Aims to disseminate software development and bureaucracy-hacking best practices across 18F and for our partners' digital service teams",
+      "billable": false,
+      "start_date": null,
+      "end_date": null,
+      "active": false
```
Comment (jessieay) [inline spec/support/fixtures/tock_projects.json]: this field is not in the real API -- is it coming soon? https://tock.18f.gov/api/projects.json

File: app/services/tock_importer.rb (hunk 2 above)
Comment (jessieay) [inline app/services/tock_importer.rb]: I think that rather than `respond_to` I'd make sure that `ClientAccount` records are being set to `active` properly

Comment (jessieay) [inline app/services/tock_importer.rb]: eg: look for a specific client account that should be active according to fixtures and make sure that on our side it is `active` too

File: spec/support/fixtures/tock_projects.json (hunk 3 above)
Comment (adelevie) [inline spec/support/fixtures/tock_projects.json]: Working on everything else until I get a new API key to Tock to see what's actually in the API.

Comment (adelevie) [inline spec/support/fixtures/tock_projects.json]: Confirmed that `active` is in the production API:
```json
		{
			"id": 120,
			"client": "General Services Administration - 18F (Non-Billable)",
			"name": "18F Branding",
			"description": "Working on branding for 18F—internal branding",
			"billable": false,
			"start_date": null,
			"end_date": null,
		}
```

**Comment by adelevie** [comment]
The production API now has token-based auth. I'll set up a user-provided service to persist the API key and update `TockImporter` to use the key.

**Comment by jessieay** [comment]
Hey @adelevie so do you want to merge this as-is since it covers #1246 and then work on token stuff separately? or do it here? getting token auth working has its own card, which I am happy for you to take over if you have time: https://github.com/18F/micropurchase/issues/951

**Pull request merged**